Solver/change_dimensions.py

```diff
@@ -211,34 +211,42 @@ def get_dimensionful_values(u_hat_dimless, boundary_conditions, nondim_disp, non
         Dimensionful displacement field in spectral space.
 
     '''
-    
     # some parameters (dimension)
     dim = len(u_hat_dimless)
-    
-        # get size of discretization
-    N = tuple(u_hat_dimless.function_space().spaces[0].bases[i].N for i in range(dim))
-    
+
+    # get size of discretization
+    N = [u_hat_dimless.function_space().spaces[0].bases[i].N
+         for i in range(dim)]
+
     # get dimensionless domain from solution
-    dom_dimless = tuple([u_hat_dimless.function_space().spaces[i].bases[i].domain for i in range(dim)])
+    dom_dimless = [u_hat_dimless.function_space().spaces[i].bases[i].domain
+                   for i in range(dim)]
 
-    # dimensionful domain  
-    dom = tuple([tuple([dom_dimless[i][j]*nondim_length for j in range(2)]) for i in range(dim)])
+    # dimensionful domain
+    dom = [
+            [
+                dom_dimless[i][j]*nondim_length for j in range(2)

```

Comment:
Why is there `for j in range(2)` and not `for j in range(dim)`?

Reply:
This refers to the start and end point of the interval so it is always equal to 2.